cicd/github-actions-oidc/deploy.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy to AWS
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Should we update this to run on PRs to be more in sync with the sls deploy command that runs on line 37?
```
sls deploy --stage ci${{ github.event.pull_request.number }}
```

Reply:
thanks, added

Reply:
Should we remove or comment this hook:
```
  push:
    branches:
      - main
```

Because the deploy step uses the [pull request number ](https://github.com/serverless-guru/templates/blob/dae8a553e2e1deb0c38cb836e414cc1db5455e4d/cicd/github-actions-oidc/deploy.yml#L40)which is probably not available in pushes

Reply:
Thanks, yes you're right.
I'll remove it.